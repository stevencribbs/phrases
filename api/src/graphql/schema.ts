import 'reflect-metadata';
import * as typeGraphQL from 'type-graphql';
import { Container } from 'typedi';

import { PingResolver } from './queries/PingResolver';
import { PhraseQueries } from './queries/PhraseQueries';
import { CreatePhraseMutation } from './mutations/CreatePhrase/CreatePhraseMutation';
import { UpdatePhraseMutation } from './mutations/UpdatePhrase/UpdatePhraseMutation';
import { DeletePhraseMutation } from './mutations/DeletePhrase/DeletePhraseMutation';
import { UserQueries } from './queries/UserQueries';
import { RegisterUserMutation } from './mutations/RegisterUser/RegisterUserMutation';
import { LoginMutation } from './mutations/Login/Login';
import { MeResolver } from './queries/MeResolver/Me';

export const buildSchema = async () => {
  return await typeGraphQL.buildSchema({
    resolvers: [
      MeResolver,
      PingResolver,
      PhraseQueries,
      UserQueries,
      CreatePhraseMutation,
      LoginMutation,
      DeletePhraseMutation,
      UpdatePhraseMutation,
      RegisterUserMutation,
    ],
    container: Container,
    // validate: { forbidUnknownValues: false },
    validate: true,
    // authChecker: checkScopes,
  });
};
