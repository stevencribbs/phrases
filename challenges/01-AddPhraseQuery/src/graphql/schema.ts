import 'reflect-metadata';
import * as typeGraphQL from 'type-graphql';
import { Container } from 'typedi';

import { PingResolver } from './queries/PingResolver';
import { PhraseQueries } from './queries/PhraseQueries';

export const buildSchema = async () => {
  return await typeGraphQL.buildSchema({
    resolvers: [PingResolver, PhraseQueries],
    container: Container,
    // validate: { forbidUnknownValues: false },
    validate: true,
  });
};
