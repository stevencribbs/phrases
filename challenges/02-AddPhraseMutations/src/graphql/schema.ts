import 'reflect-metadata';
import * as typeGraphQL from 'type-graphql';
import { Container } from 'typedi';

import { PingResolver } from './queries/PingResolver';
import { PhraseQueries } from './queries/PhraseQueries';
import { CreatePhraseMutation } from './mutations/CreatePhrase/CreatePhraseMutation';

export const buildSchema = async () => {
  return await typeGraphQL.buildSchema({
    resolvers: [PingResolver, PhraseQueries, CreatePhraseMutation],
    container: Container,
    // validate: { forbidUnknownValues: false },
    validate: true,
  });
};
