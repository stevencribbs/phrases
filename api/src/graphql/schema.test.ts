import { printSchema } from 'graphql';

import { buildSchema } from './schema';

describe('Service schema', () => {
  test('It should only change intentionally', async () => {
    const schema = await buildSchema();
    expect(printSchema(schema)).toMatchSnapshot();
  });
});
