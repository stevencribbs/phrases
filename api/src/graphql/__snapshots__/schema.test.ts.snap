// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service schema It should only change intentionally 1`] = `
"type Query {
  ping(phrase: String): String!
  phrasesForUser(userKey: String!): [Phrase!]!
  phrasesByType(phraseType: String!, userKey: String!): [Phrase!]!
  getUsers: [User!]!
  me: User
}

type Phrase {
  userKey: ID!
  phraseKey: String!
  author: String
  phraseType: String
  text: String
  tags: [String!]
  source: String
}

type User {
  userKey: String!
  firstName: String!
  lastName: String!
  name: String!
  email: String!
  confirmed: Boolean!
}

type Mutation {
  createPhrase(phrase: CreatePhraseInput!): PhraseCreatedOutput!
  updatePhrase(phrase: UpdatePhraseInput!): PhraseUpdatedOutput!
  deletePhrase(deletePhraseInput: DeletePhraseInput!): DeletePhraseResult!
  registerUser(user: RegisterUserInput!): User!
  login(password: String!, email: String!): User
  confirmUser(token: String!): Boolean!
  forgotPassword(email: String!): String!
  changePassword(data: ChangePasswordInput!): User
  logout: Boolean!
}

type PhraseCreatedOutput {
  phrase: Phrase!
}

input CreatePhraseInput {
  userKey: String!
  phraseType: String!
  author: String
  text: String
  tags: [String!]
  source: String
}

type PhraseUpdatedOutput {
  phrase: Phrase!
}

input UpdatePhraseInput {
  userKey: String!
  phraseKey: String!
  author: String
  phraseType: String
  text: String
  tags: [String!]
  source: String
}

union DeletePhraseResult = PhraseDeletedOutput | GQLError

type PhraseDeletedOutput {
  result: String!
}

type GQLError {
  message: String!
}

input DeletePhraseInput {
  userKey: String!
  phraseKey: String!
}

input RegisterUserInput {
  password: String!
  firstName: String!
  lastName: String!
  email: String!
}

input ChangePasswordInput {
  password: String!
  token: String!
}"
`;
